inst/tutorials/rescaling.Rmd

```diff
@@ -0,0 +1,290 @@
+---
+title: "Regression with rescaled variables"
+output: learnr::tutorial
+runtime: shiny_prerendered
+---
+  
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(learnr)
+```
+
+## Tutorial: Rescaling
+
+In this short tutorial, we will explore what happens when we rescale the variables in a simple bivariate regression.
+
+### Generating Token Data
+
+For the sake of this example, we will generate fake data using the following Data Generating Process:
+
+$Y = 15 + 7X + \varepsilon, \quad \varepsilon \sim \mathcal{N}(0, 10^2)$
+
+This can be done using the following code:
+
+```{r dgp}
+
+set.seed(42) #this ensures that everybody gets the same "random"" data
+
+# We'll use a sample size of n = 200
+
+x <- rnorm(n = 200, mean = 2, sd = 5) 
+# X is normally distributed w/ mean = 2 and sd = 5
+
+y <- 15 + 7*x + rnorm(n = 200, mean = 0, sd = 10)
+# the last term is the simulation of epsilon (added random noise)
+
+# store data in a data.frame
+dta <- data.frame(x = x, 
+                  y = y)
+
+plot(y~x, dta)
+```
+
+Keep an eye on the axes of the plot and the range of the data, as this tutorial is about rescaling.
+
+### Exploratory Data Analysis (EDA)
+
+Let's first print out some summary statistics for our data. This can be done by calling the `summary` function on our `data.frame`.
+
+```{r, eda-1}
+summary(dta)
+```
+
+What we are most interested with, however, is a summary of the *relationship* between $X$ and $Y$. Their correlation coefficient is an obvious candidate:
+
+```{r, eda-2}
+cor(dta)
+```
+
+The resulting output is called a "correlation matrix". The correlation between any variable and itself is obviously 1, hence the ones on the diagonal of the matrix. What we are interested are the coefficients of the off-diagonal: $Cov(X, Y) = Cov(Y, X) =$ `r cor(x, y)[1]`. This is a fairly high correlation coefficient, which is unsurprising as the data is made up.
+
+Lastly, we can try and fit $Y$ onto $X$ to see if a regression model can retrieve the underlying model from our sample.
+
+```{r, reg-1}
+fit <- lm(y~x, dta)
+summary(fit)
+```
+
+As you can see the estimated slope is `r summary(fit)$coefficients[2, 1]` and the estimated intercept is `r summary(fit)$coefficients[1, 1]`. Close enough from the "true" coefficients!
+
+```{r, reg-plot-1}
+plot(y~x, dta)
+abline(fit, col = "darkred")
+```
+
+We're now ready to explore how rescaling $X$ and $Y$ affects their relationship.
+
+### Rescaling $X$ and $Y$
+
+#### Multiplying one variable by a constant
+
+Change and run the following cell to rescale (multiply, really) $X$ and $Y$ by a constant term of your choosing, and then plot the data.
+
+```{r rescale_exo, exercise=TRUE, exercise.eval=TRUE}
+rescale_X <- 1 # change this (or not!)
+rescale_Y <- 1 # change this (or not!)
+
+dta_new <- data.frame(x = x * rescale_X,
```

Comment:
there is something wrong with the tutorial. if i try to go through it, it stops here saying it does not know what `x` is.  in general, none of the variables you define above (`dta` for example) are visible in the tutorial code boxes.
my R markdown console says
```R
Listening on http://127.0.0.1:4489
Quitting from lines 3-10 (exercise.Rmd) 
Quitting from lines 3-10 (exercise.Rmd) 
Quitting from lines 3-11 (exercise.Rmd) 
Quitting from lines 3-11 (exercise.Rmd) 
```

the tutorial runs for you?